src/crash_screen/crash_screen.c

```diff
@@ -0,0 +1,1569 @@
+#include <PR/os_internal_error.h>
+#include <stdarg.h>
+#include <string.h>
+#include "types.h"
+#include "sm64.h"
+#include "crash_screen.h"
+#include "insn_disasm.h"
+#include "map_parser.h"
+#include "audio/external.h"
+#include "buffers/framebuffers.h"
+#include "buffers/zbuffer.h"
+#include "engine/colors.h"
+#include "game/debug.h"
+#include "game/game_init.h"
+#include "game/main.h"
+#include "game/printf.h"
+#include "game/puppyprint.h"
+#include "game/rumble_init.h"
+#include "game/vc_check.h"
+
+// Crash screen font. Each row of the image fits in one u32 pointer.
+ALIGNED32 static const FontRow gCrashScreenFont[CRASH_SCREEN_FONT_CHAR_HEIGHT * CRASH_SCREEN_FONT_NUM_ROWS] = {
+    #include "textures/crash_screen/crash_screen_font.custom.ia1.inc.c"
+};
+
+#ifdef INCLUDE_DEBUG_MAP
+static struct BranchArrow sBranchArrows[DISASM_BRANCH_BUFFER_SIZE];
+static u32 sNumBranchArrows = 0;
+#endif
+
+static struct FunctionInStack sAllFunctionStack[STACK_SIZE];
+static struct FunctionInStack sKnownFunctionStack[STACK_SIZE];
+static u32 sNumKnownFunctions = 0;
+static u32 sNumShownFunctions = STACK_SIZE;
+
+static s8 sCrashScreenDirectionFlags = CRASH_SCREEN_INPUT_DIRECTION_FLAGS_NONE;
+
+static s8 sDrawCrashScreen = TRUE;
+static s8 sDrawBackground = TRUE;
+static s8 sCrashScreenSwitchedPage = FALSE;
+static s8 sCrashScreenWordWrap = TRUE;
+static s8 sStackTraceShowNames = TRUE;
+static s8 sStackTraceSkipUnknowns = FALSE;
+static s8 sAddressSelectMenuOpen = FALSE;
+static s8 sShowRamAsAscii = FALSE;
+static s8 sAddressSelecCharIndex = 2;
+static uintptr_t sAddressSelectTarget = 0;
+static uintptr_t sSelectedAddress = 0;
+static uintptr_t sScrollAddress = 0;
+static u32 sStackTraceIndex = 0;
+
+static u8 sCrashPage = PAGE_CONTEXT;
+static u8 sUpdateBuffer = TRUE;
+
+
+static const char *sCauseDesc[18] = {
+    /*EXC_INT    */ "Interrupt",
+    /*EXC_MOD    */ "TLB modification",
+    /*EXC_RMISS  */ "TLB exception on load or inst.",
+    /*EXC_WMISS  */ "TLB exception on store",
+    /*EXC_RADE   */ "Address error on load or inst.",
+    /*EXC_WADE   */ "Address error on store",
+    /*EXC_IBE    */ "Bus error on inst.",
+    /*EXC_DBE    */ "Bus error on data",
+    /*EXC_SYSCALL*/ "Failed Assert: See Assert Page",
+    /*EXC_BREAK  */ "Breakpoint exception",
+    /*EXC_II     */ "Reserved instruction",
+    /*EXC_CPU    */ "Coprocessor unusable",
+    /*EXC_OV     */ "Arithmetic overflow",
+    /*EXC_TRAP   */ "Trap exception",
+    /*EXC_VCEI   */ "Virtual coherency on inst.",
+    /*EXC_FPE    */ "Floating point exception",
+    /*EXC_WATCH  */ "Watchpoint exception",
+    /*EXC_VCED   */ "Virtual coherency on data",
+};
+
+static const char *sFpcsrDesc[6] = {
+    /*FPCSR_CE*/ "Unimplemented operation",
+    /*FPCSR_CV*/ "Invalid operation",
+    /*FPCSR_CZ*/ "Division by zero",
+    /*FPCSR_CO*/ "Overflow",
+    /*FPCSR_CU*/ "Underflow",
+    /*FPCSR_CI*/ "Inexact operation",
+};
+
+static const char *sRegNames[29] = {
+    "AT", "V0", "V1",
```

Comment:
what happened to `r0` lol (also theres 32 general registers where did the other 2 besides `r0` go)